### PR TITLE
VACMS-18544: next build feature flags

### PR DESCRIPTION
**Pull request by timcosgrove**
## Description

https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18544

- Reworks the Preview link generator so that Next Build preview checks for a feature flag being enabled, on a per-content-type basis
- Adds feature flags for every current content type in the CMS
- Adds Next.js content entities for all current Next Build content types, if they don't exist
- Fixes the port of the Next Build preview server for local development

There is a sister PR in Next Build to take advantage of this work: https://github.com/department-of-veterans-affairs/next-build/pull/676 

This PR should be merged first.

## QA steps
All QA as an administrator.

### Added feature flags
https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/admin/config/system/feature_toggle should show flags for each content type in the system, rather than just 4

### Added Next.js content entities
https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/admin/config/services/next/entity-types should show 7 configured content entities, rather than 2

### Feature flags manipulate preview functionality
1. Navigate to an Event page, i.e. https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/atlanta-health-care/events/71495
2. Hover over the 'Preview' button and note the URL. It should be of the following form: https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/node/71495?_format=static_html
3. Navigate to the Feature Toggles admin: https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/admin/config/system/feature_toggle
4. Find the feature flag `FEATURE_NEXT_BUILD_CONTENT_EVENT`. Activate it and save. 
5. Return to the Event page from 1., i.e. https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/atlanta-health-care/events/71495
6. Hover over the 'Preview' button and note the URL. It should be of a form like the following: https://next-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/api/preview?slug=/atlanta-health-care/events/71495&scope=administrator&timestamp=1723601921&secret=B6QTOkR6BmCV1nNewzyXzdMcUo-a6jJVuAFE4bFuXGc&plugin=simple_oauth&locale=en&defaultLocale=en&resourceVersion=rel%3Alatest-version
7. Click the 'Preview Button'. Currently, this will show 'Next-build server 503 response'. This is noted and will be corrected. The important thing is that the flag changes the Preview URL. 

### Listing pages only have Preview if they are Next Build enabled.
1. Navigate to https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/atlanta-health-care/events, which is an Event Listing page. You should not see a preview button. Content Build preview does not provide preview for listing pages.
2. Navigate to the Feature Toggles admin: https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/admin/config/system/feature_toggle, enable the `	FEATURE_NEXT_BUILD_CONTENT_EVENT_LISTING` flag, and save.
3.  Navigate to https://pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/atlanta-health-care/events again. You should now see a preview button, with a URL like https://next-l2wpiv91m6dwkyohkppimgnwue7btbyn.ci.cms.va.gov/api/preview?slug=/atlanta-health-care/events&scope=administrator&timestamp=1723602320&secret=vLuEm7HQi1UETFt1F-FmUQo00yJJo4Lsh7Fj14px5Fk&plugin=simple_oauth&locale=en&defaultLocale=en&resourceVersion=rel%3Alatest-version
4. Click through. You should see the 'Next-build server 503 response' error - again, this is noted and will be corrected, and is outside the scope of this ticket.

### Definition of Done

- [x] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [x] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [x] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by timcosgrove** [comment]
This comment is only to test notifications for this group, otherwise should be ignored. 

@department-of-veterans-affairs/platform-cms-helpdesk I am testing your notifications.

File: docroot/modules/custom/va_gov_preview/src/EventSubscriber/PreviewEventSubscriber.php [hunk 1]
```diff
@@ -24,11 +25,6 @@ class PreviewEventSubscriber implements EventSubscriberInterface {
 
   use StringTranslationTrait;
 
-  /**
-   * The Feature toggle name for outreach checkbox.
-   */
-  const NEXT_PREVIEW_FEATURE_NAME = 'feature_next_story_preview';
```
Comment (timcosgrove) [inline docroot/modules/custom/va_gov_preview/src/EventSubscriber/PreviewEventSubscriber.php:30]: Previously we had implemented a single flag name to control all of Next Preview. That is removed in favor of per-content-type flags.

**Review by dsasser — COMMENTED** [review]
Code looks good but there is an error saving new Feature Toggles, which I reported [here](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/18937#issuecomment-2332151210).

Despite the error, the Feature Toggle changes did persist, so I continued with QA. All QA steps passed successfully. So this could be approved, if not for the error.

**Comment by timcosgrove** [comment]
> Code looks good but there is an error saving new Feature Toggles, which I reported [here](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/18937#issuecomment-2332151210).
> 
> Despite the error, the Feature Toggle changes did persist, so I continued with QA. All QA steps passed successfully. So this could be approved, if not for the error.

The ExportFeature class was referencing `FileSystem` directly, and we have multiple file systems now. Changed it to checking Interface as type rather than the direct class. Thanks for finding!

**Review by dsasser — APPROVED** [review]
Looks good and test steps all passed, nice work!
screencapture-pr18937-l2wpiv91m6dwkyohkppimgnwue7btbyn-ci-cms-va-gov-admin-config-system-feature-toggle-2024-09-12-06_18_34

**Review by edmund-dunn — APPROVED** [review]
Nice work!

**Pull request merged**